qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -0,0 +1,195 @@
+        self.y_0 = np.eye(self.solver.model.dim)
+
+    @property
+    def target(self):
+        return self._target
+
+    @property
+    def max_circuits(self):
+        return None
+
+    def defaults(self):
+        """return backend defaults"""
+        return self._defaults
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=1024)
+
+    @staticmethod
+    def _get_info(inst):
+        p_dict = inst.operation
+        qubit = tuple(int(str(val)[-2]) for val in inst.qubits)
+        params = tuple(float(val) for val in p_dict.params)
+        return qubit, params, p_dict.name
+
+    @staticmethod
+    def _state_vector_to_result(
+        state: Union[Statevector, np.ndarray],
+        shots: int,
+        meas_return: MeasReturnType,
+        meas_level: MeasLevel,
+    ):
+        """Convert the state vector to IQ data or counts."""
+
+        # if meas_level == MeasLevel.CLASSIFIED:
+        # sample from the state vector. There might already be functions in Qiskit to do this.
+        #     counts = ...
+        # if meas_level == MeasLevel.KERNELED:
+        #     iq_data = ... create IQ data.
+        pass
+
+    def solve(self, schedule_blocks, qubits):
+        """Soleves a single schdule block and returns the unitary"""
+        if len(qubits) > 1:
+            QiskitError("TODO multi qubit gates")
+
+        signal = self.converter.get_signals(schedule_blocks)
+        time_f = schedule_blocks.duration * self.dt
+        result = self.solver.solve(
+            t_span=[0.0, time_f],
+            y0=self.y_0,
+            t_eval=[time_f],
+            signals=signal,
+            method="RK23",
+        ).y[0]
+
+        return result
+
+    # Parameters differ from overridden 'run' method
+    def run(self, run_input, **options) -> FakeJob:
+        """This should be able to return both counts and IQ data depending on the run options.
+        Loop through each circuit, extract the pulses from the circuit.calibraitons
+        and give them to the pulse solver.
+        Multiply the unitaries of the gates together to get a state vector
+        that we convert to counts or IQ Data"""
+
+        if isinstance(run_input, QuantumCircuit):
+            run_input = [run_input]
+
+        results = []
+        experiment_unitaries = {}
+
+        for circuit in run_input:
+            if circuit.calibrations.__len__ == 0:
+                raise QiskitError("TODO get schedule using pulse.InstructionScheduleMap")
+            for name, schedule in circuit.calibrations.items():
+                for (qubits, params), schedule_block in schedule.items():
+                    if (name, qubits, params) not in experiment_unitaries:
+                        experiment_unitaries[(name, qubits, params)] = self.solve(
+                            schedule_block, qubits
+                        )
+
+            psi = self.gound_state.copy()
+            for instruction in circuit.data:  # TODO check
```

Comment:
```suggestion
            for instruction in circuit.data:
```
I think this looks ok.

Reply:
I forgot to remove some of these comments before pushing it.